augur/filter/include_exclude_rules.py

```diff
@@ -178,7 +179,18 @@ def filter_by_query(metadata, query) -> FilterFunctionReturn:
     set()
 
     """
-    return set(metadata.query(query).index.values)
+    # In order to support operations such as numerical comparisons, proper data
+    # types must be set on the DataFrame columns. They are not inferred on the
+    # input metadata since it is extra overhead to do so unless necessary.
+    # The most reliable way to infer types for an existing DataFrame is to
+    # re-load the data into memory¹.
+    # ¹ https://stackoverflow.com/a/76422271
+    with StringIO() as stream:
+        metadata.to_csv(stream, index_label="id")
+        stream.seek(0)
+        metadata_copy = pd.read_csv(stream, index_col="id")
+
+    return set(metadata_copy.query(query).index.values)
```

Comment:
Could there potentially be OOM issues with this method? 
Would it be confusing to conditionalize the `dtype` param on the value of `args.query`? 

```python
dtype = object if args.query is None else None
```

Reply:
I'm not too concerned about OOM issues since metadata is already read in chunks. However, I'm more worried about the computational overhead here. I don't want to optimize for usages without `--query`, since that option is often used in workflows. I'll think about this more.

Reply:
An alternative without unnecessary I/O overhead is `pandas.to_numeric`: 77ef4a50a139a5b73272876e5488ecf8f417b1ad

Reply:
The commit linked above was merged, so this PR can skip type inference when reading metadata in filter: 322fc5961747d32f87dea5b8626db80285d748f7.